corehq/motech/openmrs/finders.py

```diff
@@ -106,6 +122,8 @@ def get_caseproperty_jsonpathvaluemap(jsonpath, value_source):
 class PropertyWeight(DocumentSchema):
     case_property = StringProperty()
     weight = DecimalProperty()
+    match_type = StringProperty(required=False, choices=MATCH_TYPES, default=MATCH_TYPE_DEFAULT)
+    match_params = ListProperty(required=False)
 
 
 class WeightedPropertyPatientFinder(PatientFinder):
```

Comment:
Not related to this PR but just wondering if you considered using ES to do this search and matching?

Reply:
:thinking: Hmmm. ... So, to flesh this out a bit more: The searching normally happens on the OpenMRS side. For example, let's say we have a case called "John Doe" born on 1 Aug. We ask OpenMRS for all the patients with the last name "Doe". Then we iterate the results, and see if any of them have the first name "John" and the birth date of 1 Aug.

But what you're suggesting, if I understand correctly, is that we store the properties we care about of *all* OpenMRS patients in ES, and then we just search for them there. (It would also help with the fact that not all properties are searchable in OpenMRS.)

I think that's a really interesting idea! It would also be useful for other integrations, like DHIS2 tracked entities. Let's chat about this more.

Reply:
Ah, I didn't realize that's how this was working. I assumed it was the other way around - like we were searching for cases matching a patient. In that case I think the way it's working is fine.